Examples/BugReportSystem/Program.cs

```diff
@@ -0,0 +1,176 @@
+    bool Validate(BugReport report);
+}
+
+public class BugReportValidation : IValidationStage
+{
+    public bool Validate(BugReport report)
+    {
+        if (string.IsNullOrEmpty(report.Title) ||
+            string.IsNullOrEmpty(report.Description) ||
+            string.IsNullOrEmpty(report.Severity) ||
+            string.IsNullOrEmpty(report.ReporterName))
+        {
+            Console.WriteLine("Bug report validation failed: Missing required fields.");
+            return false;
+        }
+        return true;
+    }
+}
+
+public interface IFormattingStage
+{
+    BugReport Format(BugReport report);
+}
+
+public class BugReportFormatting : IFormattingStage
+{
+    public BugReport Format(BugReport report)
+    {
+        report.Severity = report.Severity.ToUpper();
+
+        report.Description += $"\nTimestamp: {DateTime.Now}";
+        return report;
+    }
+}
+
+public static class BugReportConnectorFactory
+{
+    public static IBugReportConnector CreateConnector(string platform)
+    {
+        return platform switch
+        {
+            "Jira" => new JiraAdapter(new JiraBugReporter()),
+            "Slack" => new SlackAdapter(new SlackExternalService()),
+            "Discord" => new DiscordAdapter(new DiscordExternalService()),
+            _ => throw new ArgumentException("Unsupported platform")
+        };
+    }
+}
+
+public interface IDispatchStage
+{
+    void Dispatch(BugReport report);
+}
+
+public class BugReportDispatcher : IDispatchStage
+{
+    private readonly List<IBugReportConnector> _connectors;
+
+    public BugReportDispatcher(List<string> platforms)
+    {
+        _connectors = [];
+        foreach (var platform in platforms)
+        {
+            var connector = BugReportConnectorFactory.CreateConnector(platform);
+            _connectors.Add(connector);
+        }
+    }
+
+    public void Dispatch(BugReport report)
+    {
+        foreach (var connector in _connectors)
+        {
+            connector.SendBugReport(report);
+        }
+    }
+}
+
+public interface IBugReportConnector
+{
+    void SendBugReport(BugReport report);
+}
+
+public class JiraAdapter(JiraBugReporter jiraBugReporter) : IBugReportConnector
+{
+    public void SendBugReport(BugReport report)
+    {
+        jiraBugReporter.CreateIssue(report.Title, report.Description, report.Severity);
+    }
+}
+
+public class SlackAdapter(SlackExternalService slackService) : IBugReportConnector
+{
+    public void SendBugReport(BugReport report)
+    {
+        slackService.PostMessageToChannel("bug-reports", $"Bug Report: {report.Title} - {report.Description}");
+    }
+}
+
+public class DiscordAdapter(DiscordExternalService discordService) : IBugReportConnector
+{
+    public void SendBugReport(BugReport report)
+    {
+        discordService.SendMessageToChannel("bug-reports", $"Bug Report: {report.Title} - {report.Description}");
+    }
+}
+
+public class BugReport(string title, string description, string severity, string reporterName)
```

Comment:
public record BugReport(string Title, string Description, string Severity, string ReporterName);
you can use immutable object.